sim/network.go

```diff
@@ -92,12 +92,18 @@ func (nf *networkFor) RequestBroadcast(mb *gpbft.MessageBuilder) error {
 	return nil
 }
 
-func (n *Network) NetworkName() gpbft.NetworkName {
-	return n.networkName
+func (nf *networkFor) RequestSynchronousBroadcast(mb *gpbft.MessageBuilder) error {
+	msg, err := mb.Build(nf.networkName, nf.Signer, nf.ParticipantID)
+	if err != nil {
+		nf.Log("building message for: %d: %+v", nf.ParticipantID, err)
+		return err
+	}
+	nf.broadcast(msg, true)
```

Comment:
nit: combine with above in a single helper function `requestBroadcast(mb *gpbft.MessageBuilder, sync bool)`.